src/org/opensolaris/opengrok/index/Indexer.java

```diff
@@ -582,6 +583,18 @@ public static void main(String argv[]) {
                     System.exit(1);
                 }
 
+                // If the webapp is running with a config that does not contain
+                // 'projectsEnabled' property (case of upgrade or transition
+                // from project-less config to one with projects), set the property
+                // using a message so that the 'project/indexed' messages
+                // emitted during indexing do not cause validation error.
+                if (addProjects && host != null && port > 0) {
+                    Message m = Message.createMessage("config");
+                    m.addTag("set");
+                    m.setText("projectsEnabled = true");
+                    m.write(host, port);
```

Comment:
I think it is not because that completely prevents you to prepare the configuration offline.